util/interp.h

```diff
@@ -3,22 +3,139 @@
 // Copyright (c) 2018-present The Alive2 Authors.
 // Distributed under the MIT license that can be found in the LICENSE file.
 
+#include <iostream>
 #include <map>
 #include <memory>
 #include <string>
+#include <variant>
 #include <vector>
 
 namespace IR {
 class BasicBlock;
 class Function;
 class Input;
 class Value;
-}
+} // namespace IR
 
 namespace util {
 
 class ConcreteVal;
 
+struct ConcreteByte {
+  bool is_pointer;
```

Comment:
It's redundant to use this with the variant. You can use `std::holds_alternative` to check what's in the variant. Or, if you prefer, you can keep this variable and replace the variant with two separate variables.